tutorials/math/bit-manipulation.md

```diff
@@ -268,4 +290,55 @@ while (n > 0) {
     // which is same as n /= 2
     n >>= 1;
 }
+```
+
+### Two's Compliment and Negative Numbers
+
+Computers typically store integers in two’s complement representation. A positive number is represented as itself while a negative number is represented as the two’s complement of it’s absolute value (with a 1 in its sign bit to indicate that a negative value). 
+
+Let’s look at the `4-bit` integer $-3$ as an example. _If it’s a 4-bit number, we have one bit for the sign and three bits for the value_. We want the complement with respect to $2^3$, which is $8$. 
+
+The complement of $3$ (the absolute value of $-3$) with respect to $8$ is $5$. Binary value of $5$ is $0101_2$. Therefore, $-3$ in binary as a 4-bit number is $1101_2$, with the first bit being the sign bit. 
+
+In other words, the binary representation of $-K$ (negative K) as a N-bit number is $concat(1, 2^N-1 - K)$
```

Comment:
I don't get this part.

```
$concat(1, 2^N-1 - K)$
```

Reply:
Prepend (prefix) the sign bit (1) with the value for the calculated complement.

Reply:
Add it there as well

Reply:
Sure, will add it shortly.